pyproject.toml

```diff
@@ -1,7 +1,24 @@
+[project]
+name = "fenics-basix"
+packages = ["basix"]
+version = "0.7.0.dev0"
+dependencies = ["numpy>=1.21"]
+
+[project.optional-dependencies]
+ci = ["pytest-xdist", "pytest", "numba", "scipy>=0.16", "flake8", "pydocstyle", "mypy",
```

Comment:
Are all of these required? `pylit`?

Can we break the list up? Some are for running Basix tests, e.g. pytest, and other 'test' Basix, e.g. `flake8`.

Reply:
I added these incrementally to get it to pass the CI, which does need `pylit`.
I guess we can group them, e.g. `ci` `test` `lint` etc. then we have to do `pip install.[ci, test, lint]`